### PR TITLE
These are fixes for sonar scan of Python code

### PR DESCRIPTION
**Pull request by emussie**

### CODE REVIEW

File: src/DHDataset.py [hunk 1]
```diff
@@ -16,11 +16,11 @@ def __init__(self):
...
 
     @property
-    def id(self):
+    def dh_id(self):
         return self._id
 
     @id.setter
-    def id(self, value):
+    def dh_id(self, value):
         self._id = value
```
Comment (chuehlien) [inline src/DHDataset.py]: `id` and `dh_id` are both attributes of this python class so this change may introduce a bug and be in conflict with lines 74-78. I'd recommend undoing this change.

File: src/ElasticsearchDAO.py [hunk 2]
```diff
@@ -7,17 +7,17 @@ class ElasticsearchDAO(object):
     def __init__(self):
         super().__init__()
 
-    def writeToElasticsearch(self, datasets):
+    def write_To_Elasticsearch(self, datasets):
```
Comment (chuehlien) [inline src/ElasticsearchDAO.py]: The variable renames on this page and elsewhere in the PR will not pass the [sonar code smell](https://sonarcloud.io/project/issues?id=usdot-its-jpo-data-portal_datahub-ingest&open=AXE8FdPXOQUa5KFegj3H&resolved=false&types=CODE_SMELL) you are aiming to fix.

The variable renames will need to change to snake case where there's an underscore separating each word and all letters are lowercased. For example:
- writeToElasticsearch --> write_to_elasticsearch
- existingDocs --> existing_docs

File: src/ElasticsearchDAO.py [hunk 3]
```diff
@@ -7,17 +7,17 @@ class ElasticsearchDAO(object):
...
+        existing_Docs = json.loads(esresult.text)['hits']['hits']
         document = ''
         for dataset in datasets:
             found = False
-            lineObj = {}
-            lineIndexObj = {}
-            for x in existingDocs:
+            line_Obj = {}
+            line_IndexObj = {}
+            for x in existing_Docs:
```
Comment (chuehlien) [inline src/ElasticsearchDAO.py]: When you rename a variable, please make sure that the variable name is updated everywhere that it is called to make sure the rename does not result in any bugs.

For example, here lineObj is renamed to line_Obj in line 18, but various places in the script (e.g. in line 23 and 28) still use lineObj.

**Review by chuehlien — CHANGES_REQUESTED** [review]
Thanks for looking into the code smells! I added more detailed comments in the file, but generally:
1) When renaming a variable, make sure that the variable name is updated everywhere that it is called to make sure the rename does not result in any bugs.
2) The variable renames will need to change to snake case where there's an underscore separating each word and all letters are lowercased, or the introduced changes will not fix the sonar code smell you were aiming to fix. See examples in the comment.

File: src/NTLDataFormatter.py [hunk 4]
```diff
@@ -6,7 +6,7 @@
 
 class NTLDataFormatter(FormatterInterface):
     def __init__(self):
-        pass
+        pass    #this is empty explain why
```
Comment (chuehlien) [inline src/NTLDataFormatter.py]: Can change the comment to "intentionally empty constructor". Same for the comment in `src/SocrataDataFormatter.py`.

@Rodriguez-Carlos-BAH let us know if you prefer another explanation here.

**Comment by emussie** [comment]
> Thanks for looking into the code smells! I added more detailed comments in the file, but generally:
> 
> 1. When renaming a variable, make sure that the variable name is updated everywhere that it is called to make sure the rename does not result in any bugs.
> 2. The variable renames will need to change to snake case where there's an underscore separating each word and all letters are lowercased, or the introduced changes will not fix the sonar code smell you were aiming to fix. See examples in the comment.

**Comment by emussie** [comment]
It was closed accidentally

File: src/ElasticsearchDAO.py [hunk 5]
```diff
@@ -42,28 +42,28 @@ def writeToElasticsearch(self, datasets):
...
-        newLineobjDoc['dhId'] = dataset.dh_id
-        newLineobjDoc['dhLastUpdate'] = dataset.dh_last_updated
-        newLineobjDoc['dhSourceName'] = dataset.dh_source_name
-        newLineobjDoc['dhType'] = dataset.dh_type
-        newLineobjDoc['dhProjects'] = lineobjDoc['_source']['dhProjects']\
-            if ('_source' in lineobjDoc and 'dhProjects' in lineobjDoc['_source']) else []
-        newLineobjDoc['dhDataTypes'] = lineobjDoc['_source']['dhDataTypes']\
-            if ('_source' in lineobjDoc and 'dhDataTypes' in lineobjDoc['_source']) else []
-        return newLineobjDoc
+    def map_obj_doc(self, dataset, line_obj_Doc):
```
Comment (chuehlien) [inline src/ElasticsearchDAO.py]: Good work - this is looking a lot better! Remember to change `line_obj_Doc` to `line_obj_doc` (all lowercase) so that the variable name will satisfy the code smell you're trying to fix.

File: src/doi_tools.py [hunk 6]
```diff
@@ -18,9 +18,10 @@ def extractDOI(inputstring):
 
     return result
 
-def extractURLFromString(inputstring):
+def extract_url_from_string(inputstring):
     try:
         result = re.search("(?P<url>https?://[^\s]+)", inputstring).group("url").replace('\"', '')
     except:
         result = "NOT FOUND"
+        raise
```
Comment (chuehlien) [inline src/doi_tools.py]: We don't want to do a `raise` here since this is an exception that we accept. We can address this [code smell](https://sonarcloud.io/project/issues?id=usdot-its-jpo-data-portal_datahub-ingest&open=AXN7QqSfSVCEXV1-KT5R&resolved=false&severities=CRITICAL) by adding a more specific exception though. Something like "UrlNotExtractedError" could work.

Some documentation on how to create user-defined exceptions [here](https://docs.python.org/3/tutorial/errors.html#user-defined-exceptions).

File: src/test_SlackNotifier.py [hunk 7]
```diff
@@ -6,7 +6,7 @@
 
 class TestSlackNotifier(unittest.TestCase):
     @responses.activate
-    def test_sendSlackNotification(self):
+    def test_send_slack_notification(self):
```
Comment (chuehlien) [inline src/test_SlackNotifier.py]: Let's revert this change. Since this is a method to test the send_slack_notification method, it'd be more clear to leave it as is.

File: src/test_ingest.py [hunk 8]
```diff
@@ -23,20 +23,20 @@ def setUp(self):
             self.mock_data = json.load(mock_file)
 
     @responses.activate
-    def test_makeQueryCall(self):
+    def test_make_query_call(self):
```
Comment (chuehlien) [inline src/test_ingest.py]: Let's revert the change on this line as well - it's more clear as it is now.

File: src/test_ingest.py [hunk 9]
```diff
@@ -45,40 +45,40 @@ def test_ingest_ok(self, mock_writeToElasticsearch):
...
 
         ingest.ingest(test_event, test_config)
 
         mock_writeToElasticsearch.assert_called_once()
 
     @patch.object(FormatterFactory, 'get_formatter')
-    @patch.object(SlackNotifier, 'sendSlackNotification')
-    def test_ingest_invalid_formatter(self, mock_sendSlackNotification, mock_get_formatter):
+    @patch.object(SlackNotifier, 'send_slack_notification')
+    def test_ingest_invalid_formatter(self, mock_send_slack_notification, mock_get_formatter):
```
Comment (chuehlien) [inline src/test_ingest.py]: Would recommend to be consistent and revert all the `mock_send_slack_notification` to `mock_sendSlackNotification` as well.

**Review by chuehlien — APPROVED** [review]
Good work!!
